File: ontrack-repository/src/main/java/net/nemerosa/ontrack/repository/TBuildFilter.java
package net.nemerosa.ontrack.repository;

import com.fasterxml.jackson.databind.JsonNode;
import lombok.Data;

import java.util.OptionalInt;

@Data
public class TBuildFilter {

    private final OptionalInt accountId;
    private final int branchId;
    private final String name;
    private final String type;
    private final JsonNode data;

    public boolean isShared() {
        return !accountId.isPresent();
    }

}

File: ontrack-repository/src/main/java/net/nemerosa/ontrack/repository/TBuildFilter.kt
package net.nemerosa.ontrack.repository

import com.fasterxml.jackson.databind.JsonNode

data class TBuildFilter(
    val accountId: Int?,
    val branchId: Int,
    val name: String,
    val type: String,
    val data: JsonNode
) {
    val isShared: Boolean = accountId == null
}
